core/application/impl/local_key_storage.cpp

```diff
@@ -48,48 +32,110 @@ namespace kagome::application {
+      const boost::property_tree::ptree &tree) {
+    OUTCOME_TRY(sr_tree, ensure(tree.get_child_optional("sr25519keypair")));
+
+    OUTCOME_TRY(sr_pubkey_str,
+                ensure(sr_tree.get_optional<std::string>("public")));
+    OUTCOME_TRY(sr_privkey_str,
+                ensure(sr_tree.get_optional<std::string>("private")));
+
+    // get rid of 0x from beginning
+    OUTCOME_TRY(sr_pubkey_buffer, unhexWith0x(sr_pubkey_str));
+    OUTCOME_TRY(sr_privkey_buffer, unhexWith0x(sr_privkey_str));
+
+    OUTCOME_TRY(sr_pubkey,
+                crypto::SR25519PublicKey::fromSpan(sr_pubkey_buffer));
+    OUTCOME_TRY(sr_privkey,
+                crypto::SR25519SecretKey::fromSpan(sr_privkey_buffer));
+
+    sr_25519_keypair_.public_key = sr_pubkey;
+    sr_25519_keypair_.secret_key = sr_privkey;
+    return outcome::success();
   }
 
-  outcome::result<crypto::SR25519Keypair> LocalKeyStorage::loadSr25519(
-      const boost::filesystem::path &file) const {
-    using crypto::constants::sr25519::KEYPAIR_SIZE;
-    using crypto::constants::sr25519::PUBLIC_SIZE;
-    using crypto::constants::sr25519::SECRET_SIZE;
-    OUTCOME_TRY(bytes, readKeypairFromHexFile(file));
-    crypto::SR25519Keypair keypair;
-    BOOST_ASSERT(KEYPAIR_SIZE == bytes.size());
-    std::copy_n(bytes.begin(), PUBLIC_SIZE, keypair.public_key.begin());
-    std::copy_n(
-        bytes.begin() + PUBLIC_SIZE, SECRET_SIZE, keypair.secret_key.begin());
-    return keypair;
+  outcome::result<void> LocalKeyStorage::loadED25519Keys(
+      const boost::property_tree::ptree &tree) {
+    auto ed_tree_opt = tree.get_child_optional("ed25519keypair");
+    if (not ed_tree_opt) return ConfigReaderError::MISSING_ENTRY;
+    const auto &ed_tree = ed_tree_opt.value();
+
+    OUTCOME_TRY(ed_pubkey_str,
+                ensure(ed_tree.get_optional<std::string>("public")));
+    OUTCOME_TRY(ed_privkey_str,
+                ensure(ed_tree.get_optional<std::string>("private")));
+
+    // get rid of 0x from beginning
+    OUTCOME_TRY(ed_pubkey_buffer, unhexWith0x(ed_pubkey_str));
+    OUTCOME_TRY(ed_privkey_buffer, unhexWith0x(ed_privkey_str));
+
+    OUTCOME_TRY(ed_pubkey,
+                crypto::ED25519PublicKey::fromSpan(ed_pubkey_buffer));
+    OUTCOME_TRY(ed_privkey,
+                crypto::ED25519PrivateKey::fromSpan(ed_privkey_buffer));
+
+    ed_25519_keypair_.public_key = ed_pubkey;
+    ed_25519_keypair_.private_key = ed_privkey;
+    return outcome::success();
   }
 
+  outcome::result<void> LocalKeyStorage::loadP2PKeys(
+      const boost::property_tree::ptree &tree) {
+    auto p2p_tree_opt = tree.get_child_optional("p2p_keypair");
+    if (not p2p_tree_opt) return ConfigReaderError::MISSING_ENTRY;
+    const auto &p2p_tree = p2p_tree_opt.value();
+
+    OUTCOME_TRY(p2p_type,
+                ensure(p2p_tree.get_optional<std::string>("p2p_type")));
+    if (p2p_type == "ed25519") {
+      p2p_keypair_.publicKey.type = libp2p::crypto::Key::Type::Ed25519;
```

Comment:
`using KeyType = ...` makes life easier.

Reply:
Never seek for an easy way:)